physionet-django/project/management/commands/get_metrics_data.py

```diff
@@ -0,0 +1,267 @@
+                    core_project=p.core_project).latest('date')
+                # Do not update if log file is older than latest entry
+                if last_entry.date >= log_data[p.slug][0].date():
+                    continue
+            except ObjectDoesNotExist:
+                last_entry = None
+            project = Metrics.objects.create(
+                core_project=p.core_project,
+                slug=p.slug,
+                date=log_data[p.slug][0])
+            if last_entry:
+                project.running_viewcount = last_entry.running_viewcount
+            project.viewcount = log_data[p.slug][1]
+            project.running_viewcount += log_data[p.slug][1]
+            project.save()
+
+
+def validate_log_file(filename, is_cron=False, now=None):
+    """Checks if a log file is a valid NGINX log and has correct date
+
+    Checks if the given file has a date in the right place and format. If
+        called during a cron job, checks if the log file's date is the day
+        before the current date.
+
+    Args:
+        filename: An NGINX log file to parse.
+        is_cron: Optional; If True, checks if date is correct for cron job.
+        now: Optional; If running cron job, treats this as the currrent date
+            instead of timezone.now().
+
+    Returns:
+        None
+
+    Raises:
+        InvalidLogError: An error occurred obtaining date data from the log
+            file.
+        DateError: The date in the log file does not match the expected date
+            (if run as a cron job, the log file date should be the day before).
+    """
+    if now is None:
+        now = timezone.now()
+
+    with open(filename) as f:
+        first_line = f.readline()
+        split_line = first_line.split()
+        try:
+            str_date = split_line[3][1:12]
+            log_date = str_to_datetime(str_date)
+        except LookupError:
+            raise InvalidLogError
+
+    if is_cron:
+        if not (now.day - 1 == log_date.day and now.month ==
+                log_date.month and now.year == log_date.year):
+            raise DateError
+    return
+
+
+def str_to_datetime(str_date):
+    """Converts a date in string form into a Python datetime object.
+
+    Args:
+        str_date: A date in string form from an NGINX log or in DD/MM/YYYY
+            format.
+
+    Returns:
+        date: A Python datetime object.
+    """
+    abbrev_to_num = {
+        'Jan': 1,
+        'Feb': 2,
+        'Mar': 3,
+        'Apr': 4,
+        'May': 5,
+        'Jun': 6,
+        'Jul': 7,
+        'Aug': 8,
+        'Sep': 9,
+        'Oct': 10,
+        'Nov': 11,
+        'Dec': 12}
+
+    if str_date is None:
+        return None
+
+    day, month, year = str_date.split('/')
+    try:
+        month = int(month)
+    except ValueError:
+        month = abbrev_to_num[month]
+
+    date = datetime.datetime(int(year), month, int(day))
+    return date
+
+
+def str_to_bool(inp):
```

Comment:
Might be possible to use `strtobool` here? (part of Python standard library) 

```
from distutils.util import strtobool 
strtobool('true')
>> 1
strtobool('false')  
>> 0
```

Reply:
Yup, it's possible! Made those changes.